api/src/infrastructure/sql-adapter.ts

```diff
@@ -0,0 +1,74 @@
+import { WidgetDataFilters } from '@shared/dto/widgets/widget-data-filter';
+import { PageFilterQuestionMap } from '../../../shared/dto/widgets/page-filter-question-map';
+import { Injectable, Logger } from '@nestjs/common';
+
+@Injectable()
```

Comment:
Again, manipulating strings to perform queries is evil.

We are loosing type safety, and creating yet another point of truth. If we had to refactor several parts of the data model, we would be forced to edit A LOT of strings, affecting to our adaptability to change. This has to be refactored